R/database.R

```diff
@@ -747,836 +870,951 @@ get_combined_sv = function(min_vaf=0,
+#' @param limit_samples Supply this to restrict mutations to one sample.
+#' @param force_unmatched_samples Parameter description.
+#' @param basic_columns Set to TRUE to override the default behaviour of returning only the first 45 columns of MAF data.
 #' @param from_flatfile Set to TRUE to obtain mutations from a local flatfile instead of the database. This can be more efficient and is currently the only option for users who do not have ICGC data access.
-#' @param allow_clustered Logical parameter indicating whether to use SLMS-3 results with clustered events. Default is FALSE
-#' @param groups Unix groups for the samples to be included. Default is both gambl and icgc_dart samples
-#' @param include_silent Logical parameter indicating whether to include siment mutations into coding mutations. Default is TRUE
+#' @param allow_clustered Logical parameter indicating whether to use SLMS-3 results with clustered events. Default is FALSE.
+#' @param groups Unix groups for the samples to be included. Default is both gambl and icgc_dart samples.
+#' @param include_silent Logical parameter indicating whether to include siment mutations into coding mutations. Default is TRUE.
 #'
-#' @return A data frame containing all the MAF data columns (one row per mutation)
+#' @return A data frame containing all the MAF data columns (one row per mutation).
 #' @export
 #' @import tidyverse DBI RMariaDB dbplyr
 #'
 #' @examples
 #' #basic usage
-#' maf_data = get_coding_ssm(limit_cohort=c("BL_ICGC"))
-#' maf_data = get_coding_ssm(limit_samples=my_sample_ids)
+#' maf_data = get_coding_ssm(limit_cohort = c("BL_ICGC"))
+#' maf_data = get_coding_ssm(limit_samples = my_sample_ids)
+#'
 get_coding_ssm = function(limit_cohort,
                           exclude_cohort,
                           limit_pathology,
                           limit_samples,
                           force_unmatched_samples,
-                          basic_columns=TRUE,
-                          from_flatfile=FALSE,
-                          allow_clustered=FALSE,
-                          groups=c("gambl","icgc_dart"),
-                          include_silent=TRUE){
-  coding_class = c("Frame_Shift_Del",
-                   "Frame_Shift_Ins",
-                   "In_Frame_Del",
-                   "In_Frame_Ins",
-                   "Missense_Mutation",
-                   "Nonsense_Mutation",
-                   "Nonstop_Mutation",
-                   "Silent",
-                   "Splice_Region",
-                   "Splice_Site",
-                   "Targeted_Region",
-                   "Translation_Start_Site")
+                          basic_columns = TRUE,
+                          from_flatfile = FALSE,
+                          allow_clustered = FALSE,
+                          groups = c("gambl", "icgc_dart"),
+                          include_silent = TRUE){
+
   if(!include_silent){
-    coding_class=coding_class[coding_class != "Silent"]
```

Comment:
Same question as above regarding the global variable

Reply:
Same answer as above.